### PR TITLE
fr.ffs2play.ffs2play

### PR DESCRIPTION
**Pull request by liberodark**

### CODE REVIEW

**Comment by liberodark** [comment]
@gasinvein Hello sorry to disturb you but I would like to have your help on this project which is to compile the sources directly.
Having not yet understood all the operation of the flatpak I prefer to ask you rather than do something failed.
Source : https://bitbucket.org/FFS2/ffs2play/src/master/

Best Regards

**Comment by liberodark** [comment]
@bilelmoussaoui If I can have your opinion also it would be great on the methodology when it's from source.

**Comment by liberodark** [comment]
How to add 

```
  "modules": [
    {
      "name": "protobuf",
      "cleanup": [
        "protoc",
        "/bin",
        "/doc",
        "/lib/plugins"
      ],
      "sources": [
        {
          "type": "archive",
          "url": "https://github.com/protocolbuffers/protobuf/releases/download/v3.0.2/protobuf-cpp-3.0.2.tar.gz",
          "sha256": "b3732e471a9bb7950f090fd0457ebd2536a9ba0891b7f3785919c654fe2a2529"
        }
      ]
    },
```

**Comment by liberodark** [comment]
bot, build com.termius.termius

**Comment by liberodark** [comment]
bot, build fr.ffs2play.ffs2play

**Comment by liberodark** [comment]
bot, build fr.ffs2play.ffs2play

File: fr.ffs2play.ffs2play.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+app-id: fr.ffs2play.ffs2play
```
Comment (barthalion) [inline fr.ffs2play.ffs2play.yml]: Given it's open source and written in packaging-friendly language, is there any reason why it's not build from source?

File: dependencies/openssl.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (barthalion) [inline dependencies/openssl.json]: Please delete it, it's shipped by runtime.

**Comment by nedrichards** [comment]
@liberodark are upstream involved in distributing this? If not, we need to ask them! If you could put a link here to an issue or conversation doing that it'd be awesome.

Similarly, we need to send the appdata file upstream, it's generally useful for all Linux distribution types, not just flatpaks and we don't want to host 'improvements' or changes here where we can help it.

File: fr.ffs2play.ffs2play.yml (hunk 1 above)
Comment (liberodark) [reply fr.ffs2play.ffs2play.yml]: im try to build from source but im not expert of flatpak

File: dependencies/openssl.json (hunk 2 above)
Comment (liberodark) [reply dependencies/openssl.json]: ok

File: fr.ffs2play.ffs2play.yml (hunk 1 above)
Comment (liberodark) [reply fr.ffs2play.ffs2play.yml]: > @liberodark are upstream involved in distributing this? If not, we need to ask them! If you could put a link here to an issue or conversation doing that it'd be awesome.
> 
> Similarly, we need to send the appdata file upstream, it's generally useful for all Linux distribution types, not just flatpaks and we don't want to host 'improvements' or changes here where we can help it.



> @liberodark are upstream involved in distributing this? If not, we need to ask them! If you could put a link here to an issue or conversation doing that it'd be awesome.
> 
> Similarly, we need to send the appdata file upstream, it's generally useful for all Linux distribution types, not just flatpaks and we don't want to host 'improvements' or changes here where we can help it.

Yes im go to ask from the dev

File: fr.ffs2play.ffs2play.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+runtime: org.freedesktop.Platform
+runtime-version: "18.08"
+sdk: org.freedesktop.Sdk
```
Comment (LeandroStanger) [inline fr.ffs2play.ffs2play.yml]: Update runtime to 19.08

**Comment by nedrichards** [comment]
bot, build fr.ffs2play.ffs2play

File: fr.ffs2play.ffs2play.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+          size: 2669622
+
+        - type: script
+          dest-filename: apply_extra.sh
+          commands:
+            - FLATPAK_ID=fr.ffs2play.ffs2play
+            - set -e
+            - unzip FFS2-ffs2play-c6fa99db2eaf.zip
+            - pushd FFS2-ffs2play-c6fa99db2eaf/ || exit
+            - bash build-release.sh
```
Comment (gasinvein) [inline fr.ffs2play.ffs2play.yml]: Downloading sources via extra-data and building them at install time? Wat

**Comment by nedrichards** [comment]
bot, build fr.ffs2play.ffs2play

@liberodark you can run that command as well to trigger builds when you push updates.

**Pull request closed without merging**